### PR TITLE
Add metrics to scheme for containers and jobs

### PR DESCRIPTION
**Pull request by g-linville**
for acorn-io/acorn#830

### CODE REVIEW

**Comment by ibuildthecloud** [comment]
Is port always required?  I mean the schema says that, but there's no reasonable user default?

**Comment by g-linville** [comment]
> Is port always required? I mean the schema says that, but there's no reasonable user default?

@ibuildthecloud the only reasonable default I can think of is to look at the ports specified on the container itself and select the first one. I can make this field optional and implement that logic in the controller if that sounds good to you.

**Comment by ibuildthecloud** [comment]
Nah, that's probably too fancy.  Just make it required right now.

File: schema/v1/app.cue [hunk 1]
```diff
@@ -36,6 +37,7 @@ package v1
 	#WorkloadBase
 	labels: [string]:      string
 	annotations: [string]: string
+	metrics?: #Metrics
```
Comment (ibuildthecloud) [inline schema/v1/app.cue:40]: This probably belongs in WorkloadBase.

**Pull request merged**